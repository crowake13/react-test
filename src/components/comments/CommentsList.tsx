import React from 'react';
import * as RENDER_LOG from '../../constants/render-log';
import { Comment } from '../../stores/comments/comment.model';
import withRenderLog from '../shared/withRenderLog';

interface CommentListProps {
  comments: Comment[];
}

export const CommentList = ({ comments }: CommentListProps) => {
  return (
    <div className="card-body m-0">
      {comments.map((comment, index) => (
        <div key={comment.id} className={!index ? 'mb-2' : 'border-top mb-2'}>
          <h6 className="card-subtitle my-2 text-muted">{comment.email}</h6>
          <h5 className="card-title">{comment.name}</h5>
          <p className="card-text">{comment.body}</p>
        </div>
      ))}
    </div>
  );
};

export default withRenderLog({ greeting: RENDER_LOG.GREETING })(CommentList);
