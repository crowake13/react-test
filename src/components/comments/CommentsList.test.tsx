import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { Comment } from '../../stores/comments/comment.model';
import { CommentList } from './CommentsList';

let container: HTMLElement | null = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  if (!container) {
    return;
  }

  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with or without a name', () => {
  const postId = 1;
  const comments: Comment[] = [];

  act(() => {
    render(<CommentList comments={comments} />, container);
  });
  expect(container?.innerHTML).toMatchSnapshot();

  comments.push({
    postId,
    id: 1,
    name: 'title 1.1',
    email: 'email 1.1',
    body: 'body 1.1'
  });

  act(() => {
    render(<CommentList comments={comments} />, container);
  });
  expect(container?.innerHTML).toMatchSnapshot();

  comments.push({
    postId,
    id: 2,
    name: 'title 1.2',
    email: 'email 1.2',
    body: 'body 1.2'
  });

  act(() => {
    render(<CommentList comments={comments} />, container);
  });
  expect(container?.innerHTML).toMatchSnapshot();
});
