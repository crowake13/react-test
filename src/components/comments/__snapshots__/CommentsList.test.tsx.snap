// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with or without a name 1`] = `"<div class=\\"card-body m-0\\"></div>"`;

exports[`renders with or without a name 2`] = `"<div class=\\"card-body m-0\\"><div class=\\"mb-2\\"><h6 class=\\"card-subtitle my-2 text-muted\\">email 1.1</h6><h5 class=\\"card-title\\">title 1.1</h5><p class=\\"card-text\\">body 1.1</p></div></div>"`;

exports[`renders with or without a name 3`] = `"<div class=\\"card-body m-0\\"><div class=\\"mb-2\\"><h6 class=\\"card-subtitle my-2 text-muted\\">email 1.1</h6><h5 class=\\"card-title\\">title 1.1</h5><p class=\\"card-text\\">body 1.1</p></div><div class=\\"border-top mb-2\\"><h6 class=\\"card-subtitle my-2 text-muted\\">email 1.2</h6><h5 class=\\"card-title\\">title 1.2</h5><p class=\\"card-text\\">body 1.2</p></div></div>"`;
